### PR TITLE
feat(jitsi-meet): added grafana loki and otel integration for log analysis

### PR DESCRIPTION
**Pull request by ilaydadastan**
Added grafana loki and otel integration for log analysis.
This PR provides a simple and effective setup for log management and analysis using Docker, Grafana Loki and OpenTelemetry for Jitsi Meet.

### CODE REVIEW

**Comment by ltorje-8x8** [comment]
Please do not commit log files, please add .ds_store in .gitignore https://gist.github.com/lohenyumnam/2b127b9c3d1435dc12a33613c44e6308

File: docker-compose.yml [hunk 1]
```diff
@@ -459,6 +459,49 @@ services:
             - prosody
         networks:
             meet.jitsi:
+    # Log Analyser
```
Comment (saghul) [inline docker-compose.yml:462]: Please keep the whole stack in a separate yaml file and not part of the standard setup.

File: docker-compose.yml [hunk 2]
```diff
@@ -459,6 +459,49 @@ services:
...
+    grafana:
+        image: grafana/grafana:10.2.0
+        container_name: grafana
+        user: "0"
```
Comment (saghul) [inline docker-compose.yml:466]: why is this necessary?

File: docker-compose.yml [hunk 3]
```diff
@@ -459,6 +459,49 @@ services:
...
+            - ./log-analyser/otel-collector-config.yaml:/etc/otelcol-contrib/config.yaml
+            - ./log-analyser/jitsi-logs/:/tmp/jitsi-logs/
+        ports:
+            - 1888:1888 # pprof extension
+            # - 8888:8888 # Prometheus metrics exposed by the Collector
+            # - 8889:8889 # Prometheus exporter metrics
+            - 13133:13133 # health_check extension
+            - 4317:4317 # OTLP gRPC receiver
+            - 4318:4318 # OTLP http receiver
+            - 55679:55679 # zpages extension
```
Comment (saghul) [inline docker-compose.yml:503]: Ddo we really need all these ports?

**Review by saghul — COMMENTED** [review]
There are a number of stray files here, like logs and .DS_Store files, please delete those.

File: log-analyser/grafana-provisioning/.DS_Store
Comment (aaronkvanmeerten) [inline log-analyser/grafana-provisioning/.DS_Store:1]: this should not be included

File: log-analyser/grafana/grafana.db
Comment (aaronkvanmeerten) [inline log-analyser/grafana/grafana.db:1]: I think this should not be checked in?

File: log-analyser/jitsi-logs/jitsi-web.log
Comment (aaronkvanmeerten) [inline log-analyser/jitsi-logs/jitsi-web.log:1]: delete

File: log-analyser/jitsi-logs/jicofo.log
Comment (aaronkvanmeerten) [inline log-analyser/jitsi-logs/jicofo.log:1]: delete

File: log-analyser/jitsi-logs/jvb.log
Comment (aaronkvanmeerten) [inline log-analyser/jitsi-logs/jvb.log:1]: delete

File: log-analyser/jitsi-logs/prosody.log
Comment (aaronkvanmeerten) [inline log-analyser/jitsi-logs/prosody.log:1]: delete

File: log-analyser/loki/.DS_Store
Comment (aaronkvanmeerten) [inline log-analyser/loki/.DS_Store:1]: delete

**Review by aaronkvanmeerten — CHANGES_REQUESTED** [review]

File: grafana.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+version: '3.5'
+
+services:
+  # Grafana: used for visualization of metrics and log data through customizable dashboards.
+  grafana:
+    image: grafana/grafana:10.2.0
+    user: "0"
```
Comment (aaronkvanmeerten) [inline grafana.yml]: Does this need to be user 0?

File: docker-compose.yml (hunk 2 above)
Comment (ilaydadastan) [reply docker-compose.yml]: I removed container_name, I gave the user permission due to an error, so I used it.

File: docker-compose.yml (hunk 3 above)
Comment (ilaydadastan) [reply docker-compose.yml]: Since I am not currently using the specified extensions, I removed these ports.

File: log-analyser/.gitignore
Comment (ilaydadastan) [inline log-analyser/.gitignore:1]: Should I move this gitignore content to gitignore file which is in root directory or is it okay like this? @aaronkvanmeerten @ltorje-8x8

File: grafana.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      - GF_ANALYTICS_REPORTING_ENABLED=false
+    volumes:
+      - ./log-analyser/grafana:/var/lib/grafana
+      - ./log-analyser/grafana-provisioning/:/etc/grafana/provisioning/
+    ports:
+      - "3000:3000"
+    logging:
+      driver: syslog
+      options:
+        syslog-address: "udp://127.0.0.1:5123"
```
Comment (aaronkvanmeerten) [inline grafana.yml:17]: Is there a syslog listener on this port?

File: log-analyser.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+version: '3.5'
+
+services:
+  # Log Analyser: used for setting up a log analysis system for visualization, log collection and log processing.
+
+  loki:
+    container_name: loki
+    user: "0"
```
Comment (aaronkvanmeerten) [inline log-analyser.yml:8]: does this need to be user 0?

File: log-analyser/otel-collector-config.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+processors:
+  batch:
+
+exporters:
+  logging:
+    loglevel: debug
+  loki:
+    endpoint: "http://loki:3100/loki/api/v1/push"
+  otlp:
+    endpoint: tempo:4317
```
Comment (aaronkvanmeerten) [inline log-analyser/otel-collector-config.yaml:36]: Do we have a tempo endpoint here?

File: log-analyser/loki/conf/loki-config.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    - from: 2020-10-24
+      store: tsdb
+      object_store: filesystem
+      schema: v13
+      index:
+        prefix: index_
+        period: 24h
+
+ruler:
+  alertmanager_url: http://localhost:9093
```
Comment (aaronkvanmeerten) [inline log-analyser/loki/conf/loki-config.yaml:40]: This is probably not intended/needed?

File: log-analyser/README.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+- Docker
+- Docker Compose
+
+### Setup
+
+1. **Clone the repository:**
+
+    ```bash
+    git clone https://github.com/jitsi/docker-jitsi-meet.git
+    cd grafana-loki-otel
```
Comment (aaronkvanmeerten) [inline log-analyser/README.md:27]:
```suggestion
```

**Review by aaronkvanmeerten — APPROVED** [review]

**Review by aaronkvanmeerten — APPROVED** [review]
